js/modules/modules.go

```diff
@@ -123,6 +123,21 @@ type InstanceCore interface {
 
 	// sealing field will help probably with pointing users that they just need to embed this in their Instance
 	// implementations
+
+	// MakeHandledPromise needs a better name
+	// because of the way promises work and the fact that we probably don't want promises from one iteration to live
+	// till the next, this method lets a piece of module code say that they will be returning a promise that needs to be
+	// resolved/rejected within this iteration. K6 will not continue with a next iteration until either `resolve` or
+	// `reject` are called at which point the Promise usual handling of those will trigger.
+	// Caveats: this likely won't work if the Promise is rejected from within the js code
+	// This also will likely have problems with context canceling so both of those will need extra care
+	// TODO maybe export eventloop.Reserve and implement this in the js/common
+	MakeHandledPromise() (p *goja.Promise, resolve func(interface{}), reject func(interface{}))
+
+	// AddToEventLoop needs a better name
+	// MUST only be called while absolutely certain that something will not let the iteration end between the start and
+	// end of the call
+	AddToEventLoop(func())
```

Comment:
I kind of wonder if both  this and the above should not be removed and just to have something like the eventloop's `Reserve` method so that `AddToEventLoop(func() {dosomething();}` is `Reserve()(func(){dosomething();}` and `MakeHandledPromise` can be implemented in the common package the same way it's implemented currently in the `js` one (again using just eventloop#Reserve). 

We can always add more API, but removing one is a bit harder and there are more uses than just returning a Promise for telling the eventloop to not stop as you will want to add something to it. While having all 3 seems a ... bit much especially if all 3 can be implemented using only 1 of them

Reply:
I wonder also, are there any known use cases that will need this method `AddToEventLoop`? Because with the condition "something will not let the iteration end", it seems like something rather hard to use correctly :confused: I.e. it might be simpler to fall back to a `Promise` object that already has those guarantees.